gammapy/maps/wcsnd.py

```diff
@@ -495,38 +498,116 @@ def plot(self, ax=None, fig=None, add_cbar=False, stretch='linear', **kwargs):
 
-        if not self.geom.is_image:
-            raise TypeError('Only supported on 2D maps')
 
         if fig is None:
             fig = plt.gcf()
 
-        if ax is None:
-            ax = fig.add_subplot(1, 1, 1, projection=self.geom.wcs)
-
-        data = self.data
 
         kwargs.setdefault('interpolation', 'nearest')
         kwargs.setdefault('origin', 'lower')
         kwargs.setdefault('cmap', 'afmhot')
-        norm = simple_norm(data[np.isfinite(data)], stretch)
-        kwargs.setdefault('norm', norm)
 
-        caxes = ax.imshow(data, **kwargs)
 
-        cbar = fig.colorbar(caxes, ax=ax) if add_cbar else None
-        try:
-            ax.coords['glon'].set_axislabel('Galactic Longitude')
-            ax.coords['glat'].set_axislabel('Galactic Latitude')
-        except KeyError:
-            ax.coords['ra'].set_axislabel('Right Ascension')
-            ax.coords['dec'].set_axislabel('Declination')
-        except AttributeError:
-            log.info("Can't set coordinate axes. No WCS information available.")
-
-        # without this the axis limits are changed when calling scatter
-        ax.autoscale(enable=False)
-        return fig, ax, cbar
+
+        if self.geom.ndim == 2:
+            if ax is None:
+                ax = fig.add_subplot(1, 1, 1, projection=self.geom.wcs)
+
+            data = self.data
+            norm = simple_norm(data[np.isfinite(data)], stretch)
+            kwargs.setdefault('norm', norm)
+            caxes = ax.imshow(data, **kwargs)
+            cbar = fig.colorbar(caxes, ax=ax) if add_cbar else None
+            # without this the axis limits are changed when calling scatter
+            ax.autoscale(enable=False)
+
+            try:
+                ax.coords['glon'].set_axislabel('Galactic Longitude')
+                ax.coords['glat'].set_axislabel('Galactic Latitude')
+            except KeyError:
+                ax.coords['ra'].set_axislabel('Right Ascension')
+                ax.coords['dec'].set_axislabel('Declination')
+            except AttributeError:
+                log.info("Can't set coordinate axes. No WCS information available.")
+
+            return fig, ax, cbar
+
+        elif self.geom.ndim == 3:
+
+            try:
+                from ipywidgets.widgets.interaction import interact, fixed
+                import ipywidgets as widgets
+            except:
+                raise ImportError('ipywidgets is not installed.')
+
+            @interact(
+                     index=widgets.IntSlider(min=0, max=self.data.shape[0] - 1, step=1, value=1,
+                                             description=self.geom.axes_names[0]+' slice'),
+                     stretch=widgets.RadioButtons(options=['linear', 'sqrt', 'log'], value='sqrt',
+                                                  description='Plot stretch'),
+                     mapND=fixed(self)
+                    )
+            def _plot_interactive(mapND, index, stretch='linear'):
+                """
```

Comment:
For this hidden function no docstring is needed, as the user will never see it, just remove it.

Reply:
ok

Reply:
I can be nice for developpers ...!

Reply:
I think in this case the arguments are self-explaining by their name and use and the docstring does not really add any additional information. I found it rather distracting, when reading the code, because it is very unusual to have large blocks of comments on the 2nd indentation level. But I agree e.g. when a hidden function can be still used stand alone or is a hidden method on a class... but I think we do not have general policy in Gammapy, I think some hidden methods have a docstring some do not.